src/fu-engine-helper.c

```diff
@@ -187,3 +188,106 @@ fu_engine_update_devices_file(FuEngine *self, GError **error)
 	target = g_build_filename(directory, "devices.json", NULL);
 	return g_file_set_contents(target, data, (gssize)len, error);
 }
+
+static void
+fu_engine_integrity_add_measurement(GHashTable *self, const gchar *id, GBytes *blob)
+{
+	g_autofree gchar *csum = g_compute_checksum_for_bytes(G_CHECKSUM_SHA256, blob);
+	g_hash_table_insert(self, g_strdup(id), g_steal_pointer(&csum));
+}
+
+static void
+fu_engine_integrity_measure_acpi(GHashTable *self)
+{
+	g_autofree gchar *path = fu_path_from_kind(FU_PATH_KIND_ACPI_TABLES);
+	const gchar *tables[] = {"SLIC", "MSDM", "TPM2", NULL};
+
+	for (guint i = 0; tables[i] != NULL; i++) {
+		g_autofree gchar *fn = g_build_filename(path, tables[i], NULL);
+		g_autoptr(GBytes) blob = NULL;
+
+		blob = fu_bytes_get_contents(fn, NULL);
+		if (blob != NULL && g_bytes_get_size(blob) > 0) {
+			g_autofree gchar *id = g_strdup_printf("ACPI:%s", tables[i]);
+			fu_engine_integrity_add_measurement(self, id, blob);
+		}
```

Comment:
There is a problem here.  If one of these is present before and disappears the measurement is gone.  I think you want this measurement for each of these even if not present so that it's explicitly clear it was missing.

Reply:
Will you detect that they're added too from the same code?

Reply:
Yup, it's a straight port of the compare code I had before to python.

Reply:
OK that's fine to me then, thanks.